Usage guide.md

```diff
@@ -0,0 +1,73 @@
+# How to start using OneMobileSDK
+
+## Integrate SDK with your project
+
+You can integrate SDK in your project in several ways. 
+
+1. Cocoapods - look on Podfile example in this repository.
+2. Carthage - look on Cartfile example in this repository.
+3. Binary - grab packed framework directly from https://github.com/vidible/OneMobileSDK-releases/releases
+
+## Creating an `OneSDK` object
+
+TL;DR:
+```swift
+let sdk = OneSDK.Provider.default.getSDK()
+```
+
+`OneSDK` is a factory of `Player`s which share common settings and configurations. 
+But `OneSDK` itself need to be configured properly. 
+`OneSDK` should be initialized via config (`OneSDK.Configuration`) which can be received from web service. 
+
+To do so you need to request `OneSDK` future from an `OneSDK.Provider`
+
+### OneSDK.Provider
+
+This class is responsible for giving you a future to setted up OneSDK object. 
+You can use devault provider (will work for most cases) or customize it. Do to so:
+```swift
+var provider = OneSDK.Provider.default
+provider.context.extra = ["site-section": "cars"]
+
+let sdk = provider.getSDK()
+```
+Note: Contents of `extra` field is `JSON` aka `[String: Any]` 
+so you need provide something that can be eaten up by `JSONSerialization` object.
+
+### OneSDK.Context
+
+Each `OneSDK.Provider` instance contains `context` field which is used to match application and settings on web service.
+In `default` provider we use current aplication context: `OneSDK.Context.current`.
+
+This context is completed with values available in your application plist, device information, etc. 
+
+`extra` portion of context is empty by default, however, you can pass any JSON compatible dictionary here. 
+Content of this dictionary will be treated by web service in app specific way.
+
+### OneSDK.Configuration
+
+`OneSDK.Provider` have another method `func getConfiguration() -> Future<Result<OneSDK.Configuration>>`.
+
+While preferred way to construct an `OneSDK` is a provider, `OneSDK` contains initializer with `Configuration` object. 
+You can use it when you want to alter recommended settings. For example:
+```swift
+let sdk = provider.getConfiguration().map { config in 
+  var config = config
+  config.features.isHLSEnabled = false
+  
+  return OneSDK(configuration: config)
+}
+```
+
+### Note
+Content of `OneSDK.Configuration` will change from version to version 
+as we are going to more and more web oriented (and resilent!) solution.
+Please, expect code to not be stable. 
+
+Migration note
```

Comment:
in addtion we should have sections like - how it was before and how to achieve the same result with new SDK.